src/fields.jl

```diff
@@ -13,25 +13,23 @@ struct CellField{A<:AbstractArray, G<:Grid} <: Field
     grid::G
 end
 
-function CellField(::CPU, g::RegularCartesianGrid{T, <:AbstractRange}) where T <: AbstractFloat
+function CellField(::CPU, g::Grid{T}) where T
     data = zeros(T, size(g))
     CellField{typeof(data), typeof(g)}(data, g)
 end
 
-function CellField(::GPU, g::RegularCartesianGrid{T, <:AbstractRange}) where T <: AbstractFloat
```

Comment:
FYI @ali-ramadhan it's not necessary to write the second parameter of `RegularCartesianGrid` here --- Julia assumes subsequent parameters are implied. That's one reason why when creating a `struct` parameter signature, it's best to put the most oft-used parameters early in the list of parameters, e.g.

```julia
struct UsefulStruct{OftUsedP, LessUsedP, AlmostNeverUsedP}
    ...
end
```

Then

```julia
function coolfunc(s::UsefulStruct{P}) where P
    ...
end
```

works as expected.

Reply:
Ah I see, didn't know they were implied. Makes sense to list the parameters in order of importance then.

I like being explicit but I agree that maybe being too explicit just introduces lots of boilerplate.

Aisde: One idiom I think is confusing is just saying `where T` without specifying the super type of `T` when it's not `Any`. Makes it hard to figure out what the types are.

Reply:
but do we need an upper type for the `T` of grid? Why can grids only be of floats? I don't understand that conceptually.

Reply:
Ok, I guess logically `T` has to be a number. So maybe I should have just said `T <: Number` for self-documentation purposes.

Reply:
Yeah sorry I should have given an example. But yeah, I just find myself continuously going back to struct definition to figure out what the `where T` means.

Reply:
Fair.